### ISSUE TITLE
Initial state of rendered inputtext shows error

### ISSUE THREAD

**Issue by liambrozik** [AlaskaAirlines/ods-inputtext#21]
## Describe the bug
When rendering a required field onto the page, the inputtext compoents are showing an error state before the user has even interacted with them (see screenshot below).

_Edit_: I will also add, this happens even if I set `isValid="true"` as the initial state of the inputtext. It seems to be ignoring the `isValid` property.

## To Reproduce
Render the `ods-inputtext@1.0.8` component onto a page. We are using react but that doesn't seem to be the problem.

## Expected behavior
There should not be an error at the very least until the user has interacted with the field. 

Alternatively, let us handle the validation on our own. We could do this very simply if setting `required="false"` didn't automatically add "(optional)" to the input field's label. I do not understand why it automatically adds "(optional)" to the label and it seems unnecessary. 

## Screenshots
![image](https://user-images.githubusercontent.com/17190016/84321045-68e2f500-ab27-11ea-8775-c95ea6b2c3a7.png)

**Comment by geoffrey1218** [comment]
I might have introduced this with fixing #19 in 1.0.8, sorry. Install 1.0.7 for now and I'll take a look.

**Comment by liambrozik** [comment]
@blackfalcon I believe they are related.

@geoffrey1218 when I downgrade to 1.0.7 I no longer have this specific issue, but I now have the issue you are trying to fix in #19

**Comment by geoffrey1218** [comment]
I'm having trouble reproing. I'm rendering the following markup and it seems to behave as expected:
```html
<ods-inputtext label="Test input" required></ods-inputtext>
```
The required error message does not show until I enter and blur out of the input field, as expected. Tested in Svelte and React frameworks and Chrome/Firefox. 

![image](https://user-images.githubusercontent.com/4992896/84331316-b9b21800-ab3e-11ea-8e29-6f0f9e33dade.png)

Can you provide a sample of the markup you're using?

Also, note that isValid is not intended to be set by the consumer. As stated in the docs: "Can be accessed to determine if the input is in an error state or not. Not intended to be set by the consumer." It's only intended to be accessed so you can see if the input is valid or not (according to native browser validation).

As to your other point about "(optional)" being added, I believe that was in the initial spec from design. @blackfalcon, are we able to revisit that decision? Or make a prop to configure it?

**Comment by blackfalcon** [comment]
`required="false"` is incorrect. It's not `true` or `false`, it's on or off. E.g. `required` or no reference. 

```
<ods-inputtext label="First name"></ods-inputtext>
```

**Comment by liambrozik** [comment]
Here is a sample of the markup we are using: 

` <ods-inputtext ref={confirmationCodeInput} 
label="Confirmation code or ticket number" onInput={(e: any) => handleConfirmationCodeInput(e.target)} helptext="Refer to your confirmation email" error={confirmationCodeInputError} required></ods-inputtext>
`

And to the point made by @geoffrey1218 I think it would be worthwhile to reconsider adding the 
"(optional)" label to input fields that don't have the `required` attribute.

**Comment by geoffrey1218** [comment]
After pairing with Liam, the issue seems to be with the following markup:
```html
<ods-inputtext label='Test input w/ error set' required error=''></ods-inputtext>
```

The empty string comes from state within the React component. The workaround is to initialize that string to null or undefined instead, which removes the attribute altogether in the output HTML.

I'm going to close the issue since there's an easy workaround. Reconsidering adding "(optional)" to the label should be raised as a separate issue if still relevant.

**Comment by blackfalcon** [comment]
> I'm going to close the issue since there's an easy workaround. Reconsidering adding "(optional)" to the label should be raised as a separate issue if still relevant.

I am not sure how to react to this comment. The initial spec was that if `required` is not in the input, this is an optional field. 

The UX is that all fields are required unless specified as optional.

**Comment by geoffrey1218** [comment]
I went ahead and closed it because the original issue was the error state displaying before the user interacted with the field. That was happening because error was being set to empty string by the consumer; setting it to undefined instead stopped the issue from occurring. So there’s nothing to change in the component implementation for this issue. 

I felt that the discussion about optional belonged in a separate feature request since it’s separate from the issue originally reported. Apologies if I closed the issue too soon.
